commit 35a058dcb274aef9950d707bd2ba8e57da39fc47
Author: Nico Sonack <nsonack@herrhotzenplotz.de>
Date:   Sat Nov 25 12:56:28 2023 +0100

    cmd/repos: Prevent segmentation fault without configured default account when listing repositories
    
    This bug occured because the default account name wasn't checked
    for NULL causing downstream dereference of that pointer.
    
    Bug-Report: https://github.com/herrhotzenplotz/gcli/issues/118
    Reported-By: crpb (https://github.com/crpb)

diff --git a/src/cmd/repos.c b/src/cmd/repos.c
index c7581b2..4ff4285 100644
--- a/src/cmd/repos.c
+++ b/src/cmd/repos.c
@@ -372,6 +372,15 @@ subcommand_repos(int argc, char *argv[])
 		if (!owner)
 			owner = gcli_config_get_account_name(g_clictx);
 
+		/* whenever there is no default account we would be passing NULL to
+		 * gcli_get_repos. This is bad since that causes segfaults down the
+		 * line. (https://github.com/herrhotzenplotz/gcli/issues/118) */
+		if (!owner) {
+			fprintf(stderr, "error: no account specified or no default account"
+			        " configured. use -o to provide an explicit account name.\n");
+			return EXIT_FAILURE;
+		}
+
 		rc = gcli_get_repos(g_clictx, owner, n, &repos);
 		if (rc < 0)
 			errx(1, "error: failed to fetch repos: %s", gcli_get_error(g_clictx));
